src/linter.js

```diff
@@ -383,13 +386,17 @@ export default class Linter {
         return scanner.scan();
       })
       // messages should be a list of raw message data objects.
-      .then((messages) => {
-        for (let message of messages) {
+      .then(({linterMessages, scannedFiles}) => {
+        for (let message of linterMessages) {
           if (typeof message.type === 'undefined') {
             throw new Error('message.type must be defined');
           }
           this.collector._addMessage(message.type, message);
         }
+
+        for (let filename of scannedFiles) {
```

Comment:
I actually reverted the commit and I think we should do that in a separate pull request if that works for you?